src/adera/Machines/Rocket.h

```diff
@@ -111,15 +111,30 @@ class MachineRocket : public osp::active::Machine
     std::vector<osp::active::WireInput*> existing_inputs() override;
     std::vector<osp::active::WireOutput*> existing_outputs() override;
 
+    /**
+     * Return normalized power output level of the rocket this frame
+     *
+     * Returns a value [0,1] corresponding to the current output power of the
+     * engine. This value is equal to the throttle input level, unless the
+     * engine has run out of fuel, has a nonlinear throttle response, or some
+     * similar reason. Used primarily by SysExhaustPlume to determine what the
+     * exhaust plume effect should look like.
+     *
+     * @return normalized float [0,1] representing engine power output
+     */
+    float current_output_power() const
+    { return m_lastPowerOutput; }
+
 private:
     osp::active::WireInput m_wiGimbal   { this, "Gimbal"   };
     osp::active::WireInput m_wiIgnition { this, "Ignition" };
     osp::active::WireInput m_wiThrottle { this, "Throttle" };
 
     osp::active::ActiveEnt m_rigidBody  { entt::null };
     fuel_list_t m_resourceLines;
-
     Parameters m_params;
+
+    float m_lastPowerOutput{0.0f};
```

Comment:
This wouldn't really be a 'last' power output. Just use `m_powerOutput`; the values calculated in update_physics are the ones used that very frame.